auth-server/src/services/auth.ts

```diff
@@ -0,0 +1,126 @@
+import * as jwt from 'jsonwebtoken'
+import { NotFoundError } from 'objection'
+import { DataError } from 'objection-db-errors'
+import { uuid } from 'uuidv4'
+
+import User from '../model/User'
+import RefreshToken from '../model/RefreshToken'
+
+export default class AuthService {
+	constructor(){}
+
+	public async Login(username, password): Promise<any> {
+		const user = await User
+			.query()
+			.where('username', username)
+			.first()
+      
+		if (!user) { 
+			throw new NotFoundError('User not found')
+		}
+
+		const correctPassword = await user.verifyPassword(password)
+		if (!correctPassword) {
+			throw new DataError('Incorrect password')
+		}
+
+		return {
+			user: {
+				email: user.email,
+				username: user.username,
+				name: user.name
+			},
+			token: this.getSignedToken(user),
+			refreshToken: await this.getRefreshToken(user)
+		}
+	}
+
+	public async SignUp(email, password, username, name): Promise<any> {
```

Comment:
Same here, it'd be nice to be type safe.

Reply:
done